File: vulnerable-apps/swagger/safe/app/seed.py
from pathlib import Path
import hashlib

from .db import Base, SessionLocal, engine
from .models import Attachment, AuditLog, Billing, Invite, Organization, Post, Report, User

BASE_DIR = Path(__file__).resolve().parent.parent
UPLOAD_DIR = BASE_DIR / "storage" / "uploads"
SECRET_FILE = BASE_DIR / "secret.txt"


def md5_password(password: str) -> str:
    return hashlib.md5(password.encode()).hexdigest()


def main():
    Base.metadata.drop_all(bind=engine)
    Base.metadata.create_all(bind=engine)
    UPLOAD_DIR.mkdir(parents=True, exist_ok=True)
    SECRET_FILE.write_text("SWAGGER-FILE-SECRET-42d1\n", encoding="utf-8")
    db = SessionLocal()
    try:
        acme = Organization(name="Acme", slug="acme")
        globex = Organization(name="Globex", slug="globex")
        db.add_all([acme, globex])
        db.flush()

        admin = User(email="admin@bench.local", password_hash=md5_password("Admin123!"), role="admin", is_admin=True, verified=True, display_name="Acme Admin", org_id=acme.id, reset_token="admin-reset-token")
        editor = User(email="editor@bench.local", password_hash=md5_password("Editor123!"), role="editor", is_admin=False, verified=True, display_name="Acme Editor", org_id=acme.id, reset_token="editor-reset-token")
        user1 = User(email="user1@bench.local", password_hash=md5_password("User123!"), role="user", is_admin=False, verified=True, display_name="Acme User", org_id=acme.id, reset_token="user1-reset-token")
        user2 = User(email="user2@bench.local", password_hash=md5_password("User123!"), role="user", is_admin=False, verified=True, display_name="Globex User", org_id=globex.id, reset_token="globex-reset-token")
        service = User(email="admin", password_hash=md5_password("admin"), role="service", is_admin=True, verified=True, display_name="Weak Service Admin", org_id=acme.id, reset_token="service-reset-token")
        db.add_all([admin, editor, user1, user2, service])
        db.flush()

        db.add_all([
            Post(slug="acme-welcome", title="Acme welcome", body="Public Acme onboarding notes", status="published", org_id=acme.id, author_id=editor.id),
            Post(slug="acme-roadmap", title="Acme roadmap", body="Q4 launch plan", status="published", org_id=acme.id, author_id=editor.id),
            Post(slug="acme-draft", title="Acme draft", body="Draft notes", status="draft", org_id=acme.id, author_id=user1.id),
            Post(slug="globex-news", title="Globex news", body="Public Globex update", status="published", org_id=globex.id, author_id=user2.id),
            Post(slug="globex-internal", title="Globex internal", body="GLOBEX-CONFIDENTIAL-MARKER-7f3a", status="draft", org_id=globex.id, author_id=user2.id),
            Post(slug="globex-archive", title="Globex archive", body="Archived public content", status="published", org_id=globex.id, author_id=user2.id),
        ])
        db.add_all([
            Report(org_id=acme.id, name="published", query_filter="published"),
            Billing(org_id=acme.id, seats_limit=3, seats_used=0, balance_cents=10000),
            Billing(org_id=globex.id, seats_limit=3, seats_used=0, balance_cents=10000),
            Invite(org_id=acme.id, email="seed-invite@bench.local"),
            Attachment(org_id=acme.id, owner_id=user1.id, filename="welcome.txt", path=str(UPLOAD_DIR / "welcome.txt")),
            AuditLog(message="seed audit log"),
        ])
        db.commit()
    finally:
        db.close()


if __name__ == "__main__":
    main()

File: vulnerable-apps/php/vuln/app/seed.php
<?php
require_once __DIR__ . '/inc/db.php';
$pdo = pdo();
foreach (['signup_drafts', 'comments', 'posts', 'users', 'orgs'] as $table) { $pdo->exec("DROP TABLE IF EXISTS $table"); }
$pdo->exec("CREATE TABLE orgs (id INT AUTO_INCREMENT PRIMARY KEY, name VARCHAR(100), slug VARCHAR(50) UNIQUE) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4");
$pdo->exec("CREATE TABLE users (id INT AUTO_INCREMENT PRIMARY KEY, org_id INT NOT NULL, email VARCHAR(190) UNIQUE, password_hash VARCHAR(255), role VARCHAR(50), is_admin TINYINT(1) DEFAULT 0, verified TINYINT(1) DEFAULT 1, display_name VARCHAR(100), reset_token VARCHAR(100), FOREIGN KEY (org_id) REFERENCES orgs(id)) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4");
$pdo->exec("CREATE TABLE posts (id INT AUTO_INCREMENT PRIMARY KEY, org_id INT NOT NULL, author_id INT NOT NULL, slug VARCHAR(120) UNIQUE, title VARCHAR(255), body TEXT, status VARCHAR(40), FOREIGN KEY (org_id) REFERENCES orgs(id), FOREIGN KEY (author_id) REFERENCES users(id)) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4");
$pdo->exec("CREATE TABLE comments (id INT AUTO_INCREMENT PRIMARY KEY, post_id INT NOT NULL, user_id INT NOT NULL, body TEXT, FOREIGN KEY (post_id) REFERENCES posts(id), FOREIGN KEY (user_id) REFERENCES users(id)) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4");
$pdo->exec("CREATE TABLE signup_drafts (id INT AUTO_INCREMENT PRIMARY KEY, email VARCHAR(190) NOT NULL, code VARCHAR(16) NOT NULL, verified TINYINT(1) DEFAULT 0, display_name VARCHAR(100) DEFAULT '', role VARCHAR(50) DEFAULT 'user', org_slug VARCHAR(50) DEFAULT 'acme', completed TINYINT(1) DEFAULT 0) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4");
$pdo->prepare('INSERT INTO orgs (name, slug) VALUES (?, ?), (?, ?)')->execute(['Acme', 'acme', 'Globex', 'globex']);
$orgs = [];
foreach ($pdo->query('SELECT id, slug FROM orgs') as $row) { $orgs[$row['slug']] = (int)$row['id']; }
$addUser = $pdo->prepare('INSERT INTO users (org_id,email,password_hash,role,is_admin,verified,display_name) VALUES (?,?,?,?,?,?,?)');
$users = [
    [$orgs['acme'], 'admin@bench.local', 'Admin123!', 'admin', 1, 1, 'Acme Admin'],
    [$orgs['acme'], 'editor@bench.local', 'Editor123!', 'editor', 0, 1, 'Acme Editor'],
    [$orgs['acme'], 'user1@bench.local', 'User123!', 'user', 0, 1, 'Acme User'],
    [$orgs['globex'], 'user2@bench.local', 'User123!', 'user', 0, 1, 'Globex User'],
    [$orgs['acme'], 'admin', 'admin', 'service', 1, 1, 'Weak Service Admin'],
];
foreach ($users as $u) { $addUser->execute([$u[0], $u[1], password_hash($u[2], PASSWORD_DEFAULT), $u[3], $u[4], $u[5], $u[6]]); }
$userIds = [];
foreach ($pdo->query('SELECT id, email FROM users') as $row) { $userIds[$row['email']] = (int)$row['id']; }
$addPost = $pdo->prepare('INSERT INTO posts (org_id,author_id,slug,title,body,status) VALUES (?,?,?,?,?,?)');
$addPost->execute([$orgs['acme'], $userIds['editor@bench.local'], 'acme-welcome', 'Acme Welcome', 'Public Acme launch notes.', 'published']);
$addPost->execute([$orgs['acme'], $userIds['user1@bench.local'], 'acme-roadmap', 'Acme Roadmap', 'Acme internal roadmap draft.', 'draft']);
$addPost->execute([$orgs['globex'], $userIds['user2@bench.local'], 'globex-public', 'Globex Public', 'Globex public update.', 'published']);
$addPost->execute([$orgs['globex'], $userIds['user2@bench.local'], 'globex-internal', 'Globex Draft', 'GLOBEX-CONFIDENTIAL-MARKER-7f3a', 'draft']);
@mkdir('/var/www/html/uploads', 0775, true);
@mkdir('/var/www/html/downloads', 0775, true);
file_put_contents('/var/www/html/downloads/report.txt', 'public report');
?>
